src/web/Html.tsx

```diff
@@ -104,38 +133,78 @@ export const Html = React.forwardRef(
         }
       }
       // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [target])
+    }, [target, transform])
 
     const styles: React.CSSProperties = React.useMemo(
       () => ({
         position: 'absolute',
-        transform: center ? 'translate3d(-50%,-50%,0)' : 'none',
-        ...(fullscreen && {
-          top: -size.height / 2,
-          left: -size.width / 2,
+        transform: center && !transform ? 'translate3d(-50%,-50%)' : 'none',
+        ...((fullscreen || transform) && {
+          top: transform ? 0 : -size.height / 2,
+          left: transform ? 0 : -size.width / 2,
           width: size.width,
           height: size.height,
         }),
-        ...style,
+        transformStyle: transform ? 'preserve-3d' : 'initial',
+        pointerEvents: transform ? 'none' : 'auto',
+        ...(transform ? {} : style),
       }),
-      [style, center, fullscreen, size]
+      [style, center, fullscreen, size, transform]
+    )
+
+    const transformInnerStyles: React.CSSProperties = React.useMemo(
+      () => ({ position: 'absolute', pointerEvents: 'auto', ...(transform ? style : {}) }),
+      [style, transform]
     )
 
     React.useLayoutEffect(
-      () => void ReactDOM.render(<div ref={ref} style={styles} className={className} children={children} />, el)
+      () =>
+        void ReactDOM.render(
+          <div ref={transform ? transformOuterRef : ref} style={styles} className={transform ? '' : className}>
```

Comment:
Is there a reason we need to use two refs? can't we check a param and just use one ref?

Reply:
in non transform mode, you pass the forward ref; in transform mode the forward ref is put deeper into the tree, but you still need a ref to the outer element

Reply:
Ah sorry, I understand. What do you think about splitting which component we render so it's cleaner e.g:
```ts
if(transformed){
  ReactDOM.render(/* my transformed comp */ )
}
else{
  ReactDOM.render(/* my normal comp */ )
}
```

Reply:
yes it's fine